File: Hayaa.BaseModel/Hayaa.BaseModel/GridPager/BigGridPagerPamater.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Hayaa.BaseModel
{
    /// <summary>
    /// 大数据分页参数
    /// </summary>
    /// <typeparam name="T"></typeparam>
    public class BigGridPagerPamater<T> where T:class,new() 
    {
        /// <summary>
        /// 请求页
        /// </summary>
        public long Current { set; get; }
        /// <summary>
        /// 数据页尺寸
        /// </summary>
        public long PageSize { set; get; }
        /// <summary>
        /// 搜素参数
        /// </summary>
        public T SearchPamater { set; get; }
        /// <summary>
        /// 分组字段，get输出时为已排序
        /// </summary>
       public List<GroupBy> GroupBy { set; get; }
        /// <summary>
        /// 排序字段，get输出时为已排序
        /// </summary>
        public List<OrderBy> OrderBy { set; get; }
    }
}

File: Java/hayaa.basemodel/src/main/java/hayaa/basemodel/model/GridPager/BigGridPagerPamater.java
package hayaa.basemodel.model.GridPager;

import java.util.List;

public class BigGridPagerPamater<T>  {
    private long Current;
    private long PageSize;
    private T SearchPamater;
    private List<GroupBy> GroupBy;
    private List<OrderBy> OrderBy;
    public long getCurrent() {
        return Current;
    }

    public void setCurrent(long current) {
        Current = current;
    }

    public long getPageSize() {
        return PageSize;
    }

    public void setPageSize(long pageSize) {
        PageSize = pageSize;
    }

    public T getSearchPamater() {
        return SearchPamater;
    }

    public void setSearchPamater(T searchPamater) {
        SearchPamater = searchPamater;
    }

    public List<hayaa.basemodel.model.GridPager.GroupBy> getGroupBy() {
        return GroupBy;
    }

    public void setGroupBy(List<hayaa.basemodel.model.GridPager.GroupBy> groupBy) {
        GroupBy = groupBy;
    }

    public List<hayaa.basemodel.model.GridPager.OrderBy> getOrderBy() {
        return OrderBy;
    }

    public void setOrderBy(List<hayaa.basemodel.model.GridPager.OrderBy> orderBy) {
        OrderBy = orderBy;
    }


}
